test/parallel/test-http-hostname-typechecking.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+// All of these values should cause http.request() to throw synchronously
+// when passed as the value of either options.hostname or options.host
+const vals = [{}, [], NaN, Infinity, -Infinity, true, false, 1, new Date()];
+
+function errCheck(name) {
+  return new RegExp(`^TypeError: "options\\.${name}" must either be a ` +
+                    'string, undefined or null$');
+}
+
+vals.forEach((v) => {
+  assert.throws(() => http.request({hostname: v}), errCheck('hostname'));
+  assert.throws(() => http.request({host: v}), errCheck('host'));
+});
+
+// These values are OK and should not throw synchronously
+['', undefined, null].forEach((v) => {
+  assert.doesNotThrow(() => {
+    http.request({hostname: v}).on('error', common.noop);
```

Comment:
@jasnell reply on a comment above for same question

> The main part that this is testing is that the call to http.request() does not throw synchronously. The fact that it will emit the error event is non-sequential to this particular test. This is set simply to avoid the unhandled error exception.